### PR TITLE
chore: add merkle tree util function

### LINKED ISSUE #1195: chore: Update data scheme when uploading data to payout contract
### Description

When the operator wants to upload distribution onto the payout contract  with the final distribution.
The flow needs to be updated to ensure

- We generate the Merkle root and Merkle proofs using OZ library 
- Ensure the Merkle root is stored on the payout contract (current we store it as an empty string)
- Ensure the MetaPtr includes
  - the actual distribution 
  - the Merkle proof
  - the schema version

```javascript
{ 
  version: "1.0.0",
  type: "MerklePayoutStrategy",
  distribution:[],
  merkleProof: []
}

```

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

- Add util function to generate merkle tree
- Update data distribution to ensure merkle root is generated
- Add new parameters on the distribution
  - index
  - matchingAmountInToken
  - projectPayoutAddress

##### Refers/Fixes

fixes #1195

### CODE REVIEW

**Review by DanieleSalatti — APPROVED** [review]
LGTM

**Comment by nfrgosselin** [comment]
Defer to engineering review

**Comment by melissa-neira** [comment]
defer to the devs 🫡

File: packages/round-manager/src/context/round/FinalizeRoundContext.tsx [hunk 1]
```diff
@@ -225,10 +230,8 @@ async function finalizeToContract(
       payload: { finalizeRoundToContractStatus: ProgressStatus.IN_PROGRESS },
     });
 
-    const merkleRoootInBytes = ethers.utils.formatBytes32String(merkleRoot);
-
     const encodedDistribution = encodeDistributionParameters(
-      merkleRoootInBytes,
+      merkleRoot,
       distributionMetaPtr
     );
```
Comment (bhargavaparoksham) [inline packages/round-manager/src/context/round/FinalizeRoundContext.tsx:236]: @thelostone-mc  finalizeRoundToContract function which is called after encoding parameters, calls updateDistribution function via roundImplementation. 
updateDistribution has been removed in roundImp v2.  So this needs to be updated as well for this to work

**Review by bhargavaparoksham — CHANGES_REQUESTED** [review]

**Pull request merged**